medusa/process_tv.py

```diff
@@ -441,9 +441,9 @@ def unrar(self, path, rar_files, force=False):
 
                     # Skip extraction if any file in archive has previously been extracted
                     skip_extraction = False
-                    for file_in_archive in [os.path.basename(each.filename)
-                                            for each in rar_handle.infolist()
-                                            if not each.isdir()]:
+                    for file_in_archive in [os.path.basename(each)
+                                            for each in rar_handle.namelist()
+                                            if not os.path.isdir(each)]:
```

Comment:
This will always return True. You need to check the full path to the file/folder.